File: app/src/main/java/com/example/diffutilrv/repository/EmployeeRepository.java
package com.example.diffutilrv.repository;

import com.example.diffutilrv.model.Employee;

import java.util.List;

public interface EmployeeRepository {
    List<Employee> getEmployeeList();
    List<Employee> getEmployeeListSortedByRole();
    List<Employee> getEmployeeListSortedByName();
}

File: app/src/main/java/com/example/diffutilrv/repository/EmployeeRepository.kt
package com.example.diffutilrv.repository

import com.example.diffutilrv.api.ApiService
import com.example.diffutilrv.model.Employee
import kotlinx.coroutines.flow.Flow

class EmployeeRepository(
    private val api: ApiService
) {

    fun getEmployeeList(): Flow<List<Employee>> {
        return api.getEmployeeList()
    }
}